File: Task-2/task2_ws/src/age_pkg/scripts/pub_age.py
#!/usr/bin/python3

import rospy
from age_pkg.msg import Age


def pub_age():

    # node
    rospy.init_node("pub_age", anonymous=True)

    # publish
    pub = rospy.Publisher("age_topic", Age, queue_size=1)
    rate = rospy.Rate(1)

    # message
    age = Age()
    age.days = 17
    age.months = 3
    age.years = 24

    # publishing
    while not rospy.is_shutdown():
        pub.publish(age)
        rate.sleep()


if __name__ == "__main__":
    try:
        pub_age()
    except rospy.ROSInterruptException:
        pass

File: custom msg/custom msg_ws/src/age_pkg/src/pub_age.cpp
#include <ros/ros.h>
#include "age_pkg/Age.h"

using namespace ros;

int main(int argc, char **argv)
{
    // node
    init(argc, argv, "pub_age", init_options::AnonymousName);
    NodeHandle nh;

    // publish
    Publisher pub = nh.advertise<age_pkg::Age>("age_topic", 1);
    Rate rate(1);

    // message
    age_pkg::Age age;
    age.days = 17;
    age.months = 3;
    age.years = 24;

    // publishing
    while (ok())
    {
        pub.publish(age);
        spinOnce();
        rate.sleep();
    }

    return 0;
}